Setting string parameter with empty content or space only content fails from ros2cli

## Bug report

**Required Info:**

- Operating System:
  - Ubuntu 22.03
- Installation type:
  - binaries
- Version or commit hash:
  - rolling (rclpy 3.7.1, but same code is in rolling branch at github)
- DDS implementation:
  -  Fast-RTPS
- Client library (if applicable):
  - rclpy

#### Steps to reproduce issue

* in one terminal start a node (ex using python3)
```
import rclpy.node
import rclpy
rclpy.init()
node = rclpy.node.Node("mynode", allow_undeclared_parameters=True, automatically_declare_parameters_from_overrides=True)
rclpy.spin(node)
```

* in a second terminal run one of the ros2cli command
```
ros2 param set /mynode myparam " "
ros2 param set /mynode myparam "  "
ros2 param set /mynode myparam ''
ros2 param set /mynode myparam ' '
ros2 param load param.yaml
```

with param.yaml
```
/mynode:
  ros__parameters:
    myparam: " "
```

#### Expected behavior
`Set parameter successful`
At minimum for the quoted space, the value should be a quoted space, which is a valid string character, but I expect a quoted empty string to be a valid string as well.

additionally, regardless of the exception in the current behaviour, I would expect the same behaviour as a standard string  (I added parenthesis to visualize empty and spaces) 
```
> ros2 param set /mynode myparam "foo"
Set parameter successful
> ros2 param get /mynode myparam
String value is:(foo)
```
for the space and empty space
```
> ros2 param set /mynode myparam " "
Set parameter successful
> ros2 param get /mynode myparam
String value is:( )

> ros2 param set /mynode myparam ""
Set parameter successful
> ros2 param get /mynode myparam
String value is:()
```

#### Actual behavior
```
Traceback (most recent call last):
  File "/opt/ros/rolling/bin/ros2", line 33, in <module>
    sys.exit(load_entry_point('ros2cli==0.21.0', 'console_scripts', 'ros2')())
  File "/opt/ros/rolling/lib/python3.10/site-packages/ros2cli/cli.py", line 89, in main
    rc = extension.main(parser=parser, args=args)
  File "/opt/ros/rolling/lib/python3.10/site-packages/ros2param/command/param.py", line 39, in main
    return extension.main(args=args)
  File "/opt/ros/rolling/lib/python3.10/site-packages/ros2param/verb/set.py", line 61, in main
    parameter.value = get_parameter_value(string_value=args.value)
  File "/opt/ros/rolling/local/lib/python3.10/dist-packages/rclpy/parameter.py", line 230, in get_parameter_value
    value.string_value = yaml_value
  File "/opt/ros/rolling/local/lib/python3.10/dist-packages/rcl_interfaces/msg/_parameter_value.py", line 239, in string_value
    assert \
AssertionError: The 'string_value' field must be of type 'str'
```

OR for the load
```
Traceback (most recent call last):
  File "/opt/ros/rolling/bin/ros2", line 33, in <module>
    sys.exit(load_entry_point('ros2cli==0.21.0', 'console_scripts', 'ros2')())
  File "/opt/ros/rolling/lib/python3.10/site-packages/ros2cli/cli.py", line 89, in main
    rc = extension.main(parser=parser, args=args)
  File "/opt/ros/rolling/lib/python3.10/site-packages/ros2param/command/param.py", line 39, in main
    return extension.main(args=args)
  File "/opt/ros/rolling/lib/python3.10/site-packages/ros2param/verb/load.py", line 53, in main
    load_parameter_file(node=node, node_name=node_name, parameter_file=args.parameter_file,
  File "/opt/ros/rolling/lib/python3.10/site-packages/ros2param/api/__init__.py", line 49, in load_parameter_file
    future = client.load_parameter_file(parameter_file, use_wildcard)
  File "/opt/ros/rolling/local/lib/python3.10/dist-packages/rclpy/parameter_client.py", line 307, in load_parameter_file
    param_dict = parameter_dict_from_yaml_file(parameter_file, use_wildcard)
  File "/opt/ros/rolling/local/lib/python3.10/dist-packages/rclpy/parameter.py", line 319, in parameter_dict_from_yaml_file
    return _unpack_parameter_dict(namespace, param_dict)
  File "/opt/ros/rolling/local/lib/python3.10/dist-packages/rclpy/parameter.py", line 341, in _unpack_parameter_dict
    parameter.value = get_parameter_value(str(param_value))
  File "/opt/ros/rolling/local/lib/python3.10/dist-packages/rclpy/parameter.py", line 230, in get_parameter_value
    value.string_value = yaml_value
  File "/opt/ros/rolling/local/lib/python3.10/dist-packages/rcl_interfaces/msg/_parameter_value.py", line 239, in string_value
    assert \
AssertionError: The 'string_value' field must be of type 'str'
```
#### Additional information

The culprit of this behavior is `yaml.safe_load()` that will generate a `NoneType` for "", '', " ", ' ' 
```
type(yaml.safe_load(" "))
<class 'NoneType'>
```
I can understand that strings must be passed through `yaml.safe_load`, but if the result is _NoneType_, it means that the `safe_load` did not generate a _ParseError_ or any other exception, and thus the _string_value_ should be used as is.
I tried that solution
`_value.string_value = yaml_value if yaml_value is not None else string_value`

which sets the value correctly, when the content is empty or space(s) 

( I must mention that set the parameter once does not set the parameter type, setting it twice does (for any `ros2 param set` call on allow_undeclared_parameters=True + automatically_declare_parameters_from_overrides=True )  but this is not related to that issue)

--- a/rclpy/rclpy/parameter.py
+++ b/rclpy/rclpy/parameter.py
@@ -222,7 +222,7 @@ def get_parameter_value(string_value: str) -> ParameterValue:
             value.string_value = string_value
     else:
         value.type = ParameterType.PARAMETER_STRING
-        value.string_value = yaml_value
+        value.string_value = yaml_value if yaml_value is not None else string_value
     return value
 
 

--- a/rclpy/test/test_parameter.py
+++ b/rclpy/test/test_parameter.py
@@ -21,10 +21,10 @@ import pytest
 from rcl_interfaces.msg import Parameter as ParameterMsg
 from rcl_interfaces.msg import ParameterType
 from rcl_interfaces.msg import ParameterValue
+from rclpy.parameter import get_parameter_value
 from rclpy.parameter import Parameter
 from rclpy.parameter import parameter_dict_from_yaml_file
 from rclpy.parameter import parameter_value_to_python
-from rclpy.parameter import get_parameter_value
 
 
 class TestParameter(unittest.TestCase):
@@ -207,7 +207,7 @@ class TestParameter(unittest.TestCase):
             try:
                 p = get_parameter_value(str(input_value))
             except Exception as e:
-                assert False, f"failed to get param_value, reason: {e}"
+                assert False, f'failed to get param_value, reason: {e}'
             self.assertEqual(p, expected_value)
 
     def test_parameter_value_to_python(self):

--- a/rclpy/test/test_parameter.py
+++ b/rclpy/test/test_parameter.py
@@ -24,6 +24,7 @@ from rcl_interfaces.msg import ParameterValue
 from rclpy.parameter import Parameter
 from rclpy.parameter import parameter_dict_from_yaml_file
 from rclpy.parameter import parameter_value_to_python
+from rclpy.parameter import get_parameter_value
 
 
 class TestParameter(unittest.TestCase):
@@ -166,6 +167,48 @@ class TestParameter(unittest.TestCase):
             name='double_array',
             value=ParameterValue(type=8, double_array_value=[1.0, 2.0, 3.0])
         ))
+
+    def test_get_parameter_value(self):
+        """Test the get_parameter_value function."""
+        test_cases = [
+            (True, ParameterValue(type=int(ParameterType.PARAMETER_BOOL), bool_value=True)),
+            (42, ParameterValue(type=int(ParameterType.PARAMETER_INTEGER), integer_value=42)),
+            (3.5, ParameterValue(type=int(ParameterType.PARAMETER_DOUBLE), double_value=3.5)),
+            ('foo', ParameterValue(type=int(ParameterType.PARAMETER_STRING), string_value='foo')),
+            (' ', ParameterValue(type=int(ParameterType.PARAMETER_STRING), string_value=' ')),
+            ('', ParameterValue(type=int(ParameterType.PARAMETER_STRING), string_value='')),
+            (
+                [True, False],
+                ParameterValue(
+                    type=int(ParameterType.PARAMETER_BOOL_ARRAY),
+                    bool_array_value=[True, False])
+            ),
+            (
+                [1, 2, 3],
+                ParameterValue(
+                    type=int(ParameterType.PARAMETER_INTEGER_ARRAY),
+                    integer_array_value=[1, 2, 3])
+            ),
+            (
+                [1.0, 2.0, 3.0],
+                ParameterValue(
+                    type=int(ParameterType.PARAMETER_DOUBLE_ARRAY),
+                    double_array_value=[1.0, 2.0, 3.0])
+            ),
+            (
+                ['foo', 'bar'],
+                ParameterValue(
+                    type=int(ParameterType.PARAMETER_STRING_ARRAY),
+                    string_array_value=['foo', 'bar'])
+            ),
+        ]
+
+        for input_value, expected_value in test_cases:
+            try:
+                p = get_parameter_value(str(input_value))
+            except Exception as e:
+                assert False, f"failed to get param_value, reason: {e}"
+            self.assertEqual(p, expected_value)
 
     def test_parameter_value_to_python(self):
         """Test the parameter_value_to_python conversion function."""

